sources/api/netdog/build.rs

```diff
@@ -4,8 +4,31 @@ use std::env;
 use std::fs::File;
 use std::io::Write;
 use std::path::PathBuf;
+use std::process;
 
 fn main() {
+    // The code below emits `cfg` operators to conditionally compile this program based on the
+    // current variant.
+    // TODO: Replace this approach when the build system supports ideas like "variant
+    // tags": https://github.com/bottlerocket-os/bottlerocket/issues/1260
+    println!("cargo:rerun-if-env-changed=VARIANT");
+    if let Ok(variant) = env::var("VARIANT") {
+        if variant.starts_with("aws") {
+            println!("cargo:rustc-cfg=bottlerocket_platform=\"aws\"");
+        } else if variant.starts_with("vmware") {
+            println!("cargo:rustc-cfg=bottlerocket_platform=\"vmware\"");
+        } else if variant.starts_with("metal") {
+            println!("cargo:rustc-cfg=bottlerocket_platform=\"metal\"");
+        } else {
+            eprintln!(
+            "For local builds, you must set the 'VARIANT' environment variable so we know which data \
+            provider to build. Valid values are the directories in models/src/variants/, for \
+            example 'aws-ecs-1'."
+            );
+            process::exit(1);
+        }
```

Comment:
I had to do something similar somewhere else (not yet in the code), what do you think about?

```suggestion
let platform = variant.split('-').collect::<Vec<&str>>()[0].to_string();
println!("cargo:rustc-cfg=bottlerocket_platform=\"{}\"", platform);
```

Reply:
It's definitely less code, and I like the fact that we don't have to specify all the variants, but I could see a few issues.
* We don't get a nice error message if someone misspells the variant - we get a more cryptic compiler message
* I think it's reasonable to assume that a variant name wouldn't be zero-length, but we get the same issue as above where the build will happen with `cargo:rustc-cfg=bottlerocket_platform="awsk8s121" and we'd get a less than friendly error.

Reply:
- For 1), in the current implementation I can do `aws-not-existing` and get a compiler error
- For 2), in the current implementation, `awsk8s121` will cause the same error as 1)

So the question is how much validation do we want to do while parsing the variant's name to provide a friendly error?, or else we have to trade off some errors for others when the user passes an invalid variant.

Reply:
I implemented your suggestion, and opted to add a `compile_error!()` after a `cfg` directive - see `wicked.rs`